charon/src/translate/translate_types.rs

```diff
@@ -601,12 +603,9 @@ impl<'tcx, 'ctx, 'ctx1> BodyTransCtx<'tcx, 'ctx, 'ctx1> {
                     }
                 };
 
-                // Translate the span information
-                let span = self.translate_span_from_rspan(field_def.span);
-
                 // Store the field
                 let field = Field {
-                    span,
+                    item_meta: item_meta.clone(),
```

Comment:
You're using the meta information of the *type* here, but you should use the type information from the *field*, which is different.
Ex.:
```rust
#[rename("Type")]
struct S {
  #[rename("y")]
  x : u32
}
```